### PR TITLE
Add a proxy_requests section and various fixes/updating wording

### LINKED ISSUE #750: Guide: Using proxy requests to monitor networking devices
The Sensu Core 2.0 Monitoring External Resources guide should include a section on using proxy resources to monitor networking devices.

Content notes:

- Use `sensuctl create` (This should be `sensuctl entity create`.)

### PR DESCRIPTION
**Pull request by apaskulin**
## Description
- Updates the guide to monitoring external resources to include using proxy requests
- Updates the reference docs for proxy requests

## Motivation
Closes #750, replacement for https://github.com/sensu/sensu-docs/pull/811

## Review Instructions
Guide: https://sensu-docs-stage.herokuapp.com/sensu-go/5.0/guides/monitor-external-resources/
Reference: https://sensu-docs-stage.herokuapp.com/sensu-go/5.0/reference/checks/#proxy-requests

### CODE REVIEW

**Comment by rgeniesse** [comment]
This will need updates following changes in beta 8.

File: content/sensu-go/5.0/reference/checks.md [hunk 1]
```diff
@@ -387,11 +398,10 @@ example      | {{< highlight shell >}}"splay": true{{< /highlight >}}
 
 |splay_coverage  | |
 -------------|------
-description  | The splay coverage percentage use for proxy check request splay calculation. The splay coverage is used to determine the amount of time check requests can be published over (before the next check interval).
-required     | false
+description  | The **percentage** of the check interval over which Sensu can execute the check for all applicable entities, as defined in the entity attributes. Sensu uses the splay coverage attribute to determine the amount of time check requests can be published over (before the next check interval).
```
Comment (apaskulin) [inline content/sensu-go/5.0/reference/checks.md:411]: Needs an example

**Comment by cwjohnston** [comment]
@sensu/reliability please review via staging links in the original post.

File: content/sensu-go/5.0/guides/monitor-external-resources.md [hunk 2]
```diff
@@ -10,107 +10,290 @@ menu:
...
-entity, and the result of each check request will be reported under the matched
-entity and not the agent's entity that emitted the result.
-
-## Why use proxy requests and entities?
-
-Sensu proxy requests and entities allow Sensu to monitor external resources
+Proxy entities allow Sensu to monitor external resources
 on systems or devices where a Sensu agent cannot be installed, like a
 network switch or a website.
+You can create [proxy entities][1] using [sensuctl][8], the [Sensu API][9], or the [`proxy_entity_name` check attribute][2]. When executing checks that include a `proxy_entity_name`, Sensu agents report the resulting event under the proxy entity instead of the agent entity.
```
Comment (cwjohnston) [inline content/sensu-go/5.0/guides/monitor-external-resources.md:19]: Results are also associated with proxy entities when the check specifies values for `proxy_request` attribute.

File: content/sensu-go/5.0/guides/monitor-external-resources.md [hunk 3]
```diff
@@ -10,107 +10,290 @@ menu:
...
 {{< highlight shell >}}
 sudo curl https://raw.githubusercontent.com/sensu/sensu-go/60e6a68aecb0c8e0c2dc51714e08462eb81b4413/examples/checks/http_check.sh \
 -o /usr/local/bin/http_check.sh && \
 sudo chmod +x /usr/local/bin/http_check.sh
 {{< /highlight >}}
 
-While this command is appropriate when running a few agents, you should consider
-using a **configuration management** tool or use [Sensu assets][5] to provide
-runtime dependencies to checks on bigger environments.
+_PRO TIP: While this command is appropriate when running a few agents, you should consider
```
Comment (cwjohnston) [inline content/sensu-go/5.0/guides/monitor-external-resources.md]:
```suggestion
_PRO TIP: While this command may be appropriate when running a few agents, you should consider
```

File: content/sensu-go/5.0/guides/monitor-external-resources.md [hunk 4]
```diff
@@ -10,107 +10,290 @@ menu:
...
+using [Sensu assets][5] or a [configuration management][15] tool to provide
+runtime dependencies for checks in bigger environments._
```
Comment (cwjohnston) [inline content/sensu-go/5.0/guides/monitor-external-resources.md]:
```suggestion
runtime dependencies._
```

File: content/sensu-go/5.0/guides/monitor-external-resources.md [hunk 5]
```diff
@@ -10,107 +10,290 @@ menu:
...
-Now that our script is installed, the second step is to create a check named
-`check-sensuapp`, which runs the command `http_check.sh https://sensu.io`, at an
-**interval** of 60 seconds, for all entities subscribed to the `sensu-app`
-subscription, using the `sensu.io` proxy entity name.
+Now that our script is installed, we'll create a check named
+`check-http`, which runs the command `http_check.sh https://sensu.io`, at an
+interval of 60 seconds, for all entities subscribed to the `proxy`
+subscription, using the `sensu-site` proxy entity name.
+
+Create a file called `check.json` and add the following check definition.
```
Comment (cwjohnston) [inline content/sensu-go/5.0/guides/monitor-external-resources.md:49]:
```suggestion
Create a file called `check.json` with the following check definition as it's content.
```

File: content/sensu-go/5.0/guides/monitor-external-resources.md [hunk 6]
```diff
@@ -10,107 +10,290 @@ menu:
...
-### Installing a script
+### Installing an HTTP check script
 
-We will use a [bash script][4], named `http_check.sh`, to perform an HTTP
-request using **curl**.
+First, we'll install a [bash script][4], named `http_check.sh`, to perform an HTTP
+check using **curl**.
 
 {{< highlight shell >}}
 sudo curl https://raw.githubusercontent.com/sensu/sensu-go/60e6a68aecb0c8e0c2dc51714e08462eb81b4413/examples/checks/http_check.sh \
```
Comment (cwjohnston) [inline content/sensu-go/5.0/guides/monitor-external-resources.md]: We can use a shorter URL if we reference a tagged commit, e.g. `https://raw.githubusercontent.com/sensu/sensu-go/2.0.0-beta.8-1/examples/checks/http_check.sh` works

I hope engineering will be retroactively tagging GA and newer releases in the sensu-go repo soon, so we could use a non-beta tag.

**Review by rgeniesse — APPROVED** [review]
Tested on CentOS and Ubuntu, 👍.

File: content/sensu-go/5.0/guides/monitor-external-resources.md (hunk 6 above)
Comment (cwjohnston) [reply content/sensu-go/5.0/guides/monitor-external-resources.md]: Tags have been backfilled!

```suggestion
sudo curl https://raw.githubusercontent.com/sensu/sensu-go/5.1.0/examples/checks/http_check.sh \
```

**Review by cwjohnston — CHANGES_REQUESTED** [review]

**Comment by apaskulin** [comment]
Thanks to everyone who helped test and review this, especially @rgeniesse for the opening the original PR. I think this is a solid improvement over the existing guide, so I'd like to merge this for now and handle additional improvements in separate issues.

**Pull request merged**